addons/mail/static/src/core/common/message.js

```diff
@@ -134,16 +134,16 @@ export class Message extends Component {
         );
         useEffect(
             () => {
-                if (!this.shadowRoot) {
+                if (!this.shadowRoot || !this.message.type || this.message.type.includes("email")) {
                     return;
                 }
-                if (this.state.originalFormat) {
-                    this.shadowRoot.removeChild(this.shadowStyle);
-                } else {
+                if (cookie.get("color_scheme") === "dark") {
                     this.shadowRoot.insertBefore(this.shadowStyle, this.shadowRoot.firstChild);
+                } else {
+                    this.shadowRoot.removeChild(this.shadowStyle);
                 }
             },
-            () => [this.state.originalFormat]
+            () => [this.message?.type, this.state.shadowMounted]
```

Comment:
Note that `useEffect` is kept because in practice the `useEffect` might be called before `onMounted`, and it's possible that message being mounted doesn't have a `type` (yet).

Reply:
In which situation can `useEffect` be called before `onMounted`? This seems really strange.

Reply:
I don't think there's anything in OWL to enforce that `useEffect` is always called after `onMounted`